src/components/launchermodel.cpp

```diff
@@ -726,7 +745,13 @@ LauncherItem *LauncherModel::addItemIfValid(const QString &path)
     LauncherItem *item = new LauncherItem(path, this);
 
     bool isValid = item->isValid();
-    bool shouldDisplay = item->shouldDisplay();
+    bool shouldDisplay = item->shouldDisplay() && _categories.isEmpty();
```

Comment:
Correct is debatable, but it is intentional.  We have applications that we don't want to appear in the launcher normally, but do want to appear if listed by category.  If there's another use case for settings NoDisplay=true and specifying a category, then I guess we should add a property to the model to ignore NoDisplay but I can't really think of one.